projects/storefrontlib/src/cms-components/myaccount/payment-methods/payment-methods.service.ts

```diff
@@ -0,0 +1,121 @@
+import { Injectable } from '@angular/core';
+import { combineLatest, Observable } from 'rxjs';
+import { map, tap } from 'rxjs/operators';
+import {
+  PaymentDetails,
+  TranslationService,
+  UserPaymentService,
+} from '@spartacus/core';
+import { Card } from '../../../shared';
+import { ICON_TYPE } from '../../misc';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class PaymentMethodsService {
+  loading$: Observable<
+    boolean
+  > = this.userPaymentService.getPaymentMethodsLoading();
+  editCard: string = null;
+
+  constructor(
+    protected userPaymentService: UserPaymentService,
+    protected translation: TranslationService
+  ) {
+    this.userPaymentService.loadPaymentMethods();
+  }
+
+  get(): Observable<PaymentDetails[]> {
+    return this.userPaymentService.getPaymentMethods().pipe(
```

Comment:
We typically create a local (protected) property to avoid recreating an observable when the component is recreating (and calls the `get`). Then, you could use the `defer()` operator in the property to avoid the observable being created upfront.